spec/requests/api/suppliers_controller_locations_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::SuppliersController do
+  subject(:get_supplier_locations) { get "/api/v1/suppliers/#{supplier_id}/locations#{query_params}", headers: headers, params: {} }
+
+  let(:access_token) { 'spoofed-token' }
+  let(:headers) { { 'Authorization': "Bearer #{access_token}" } }
+  let(:response_json) { JSON.parse(response.body) }
+  let(:query_params) { '' }
+
+  describe 'GET /suppliers/:supplier_id/locations' do
+    let(:schema) { load_yaml_schema('get_locations_responses.yaml') }
+
+    let(:supplier) { create(:supplier) }
+    let(:supplier_id) { supplier.id }
+    let(:location_a) { create(:location, suppliers: [supplier], key: 'key_aaa') }
+    let(:location_b) { create(:location, suppliers: [supplier], key: 'key_bbb') }
+
+    it_behaves_like 'an endpoint that responds with success 200' do
+      before do
+        create(:move, :requested, supplier: supplier)
```

Comment:
I added this because I wanted to test the schema with a proper response (and not empty array)

We could achieve the same with 'let!', but we try to avoid it.